Add X500 Lidar Models

Claudio-Chies opened the pull request:
Issue Tracked in https://github.com/PX4/PX4-Autopilot/issues/23602


This PR Adds two X500 models, one with a front facing rangefinder for collision prevention, and one down facing lidar for normal rangefinder use case testing. 
For the model and specs, i used a Lightware LW20/C
I also renamed the `x500_lidar` to `x500_lidar_2d`, to better differentiate the three
#### Changes:
**New Models:**
`x500_lidar_front`
`x500_lidar_front`

<img src="https://github.com/user-attachments/assets/89342621-fdd8-4742-9894-8a347c733e98" height=200 />
<img src="https://github.com/user-attachments/assets/d6db065c-7810-4018-9a82-f91e9f701cb9" height=200 />

**changes:**
`x500_lidar` --> `x500_lidar_2d`
moved from `gpu_ray` to `gpu_lidar`, as  `gpu_ray` will be [deprecated](http://sdformat.org/spec?ver=1.11&elem=sensor#sensor_type) 




## Parent PR's
- [ ] https://github.com/PX4/PX4-Autopilot/pull/23879

dakejahl commented:
It might make more sense to create a distance_sensor model and then attach that with different orientations. Like in this PR 
 https://github.com/PX4/PX4-gazebo-models/pull/51/files#diff-b583a7cbffcfe5cd3e7660c84fa84e9c9705d39dd684fc521a623041c1f04e1f

I did this a while back to test optical flow with gz
https://github.com/PX4/PX4-Autopilot/pull/23456

Claudio-Chies commented:
@dakejahl yea, that would be nicer.
my starting point was the broadcom optical flow lidar, which was also implemented this way. The problem i had was that the orientation in the gazebo message is only from the '<pose>' in the '<sensor>' tag, and not the orientation from the sensor in respect of the body. Meaning i dont get the sensor to body orientation in the GZBridge. Which is why i implemented it this way. 
This way i can then set the orientation enum in the UORB message based on the quaterion sent in the gz message.

It might be that i missed something and its actually possible. But i couldnt get it working.
